app/controllers/events_controller.rb

```diff
@@ -93,4 +93,8 @@ def destroy
   def event_params
     params.require(:event).permit(:name, :description, :occurs_at, :prod_description, :twitter, :url, :picture, :picture_cache)
   end
+
+  def load_user_event!(slug)
+    current_user.events.where(slug: slug).first!
```

Comment:
my fault here. it's better find_by(slug: sug) than this where().first
it returns the first and faster, and it's more elegant.